support/hololens/ServoApp/ServoControl/ServoControl.cpp

```diff
@@ -93,15 +96,50 @@ void ServoControl::OnSurfaceManipulationDelta(
   e.Handled(true);
 }
 
-void ServoControl::OnSurfaceClicked(IInspectable const &,
-                                    Input::TappedRoutedEventArgs const &e) {
+void ServoControl::OnSurfaceTapped(IInspectable const &,
+                                   Input::TappedRoutedEventArgs const &e) {
   auto coords = e.GetPosition(Panel());
   auto x = coords.X * mDPI;
   auto y = coords.Y * mDPI;
   RunOnGLThread([=] { mServo->Click(x, y); });
   e.Handled(true);
 }
 
+void ServoControl::OnSurfacePointerPressed(
+    IInspectable const &, Input::PointerRoutedEventArgs const &e) {
+  if (e.Pointer().PointerDeviceType() ==
+      Windows::Devices::Input::PointerDeviceType::Mouse) {
+    auto point = e.GetCurrentPoint(Panel());
+
+    auto x = point.Position().X * mDPI;
+    auto y = point.Position().Y * mDPI;
+    auto props = point.Properties();
+    std::optional<Servo::MouseButton> b = {};
+
+    if (props.IsLeftButtonPressed()) {
+      b = Servo::MouseButton::Left;
+    } else if (props.IsRightButtonPressed()) {
+      b = Servo::MouseButton::Right;
+    } else if (props.IsMiddleButtonPressed()) {
+      b = Servo::MouseButton::Middle;
+    }
+
+    if (!b.has_value() && mPressedMouseButton.has_value()) {
+      auto b = *mPressedMouseButton;
+      mPressedMouseButton = {};
+      RunOnGLThread([=] { mServo->MouseUp(x, y, b); });
+      e.Handled(true);
+    }
+
+    if (b.has_value()) {
```

Comment:
Won't this also trigger when we've just run a MouseUp event in the previous branch?

Reply:
It won't. I think the confusion comes from the very non descriptive variable name (`b`) which is used twice. I will change that.